src/master.py

```diff
@@ -120,31 +140,19 @@ def remaster(stackname, new_master_stackname):
     buildvars.refresh(stackname, context)
 
     # remove knowledge of old master
-    def work():
+    def workerfn():
         remote_sudo("rm -f /etc/salt/pki/minion/minion_master.pub")  # destroy the old master key we have
     LOG.info("removing old master key from minion")
-    core.stack_all_ec2_nodes(stackname, work, username=config.BOOTSTRAP_USER)
+    core.stack_all_ec2_nodes(stackname, workerfn, username=config.BOOTSTRAP_USER)
 
     # update ec2 nodes
     LOG.info("updating nodes")
     bootstrap.update_ec2_stack(stackname, context, concurrency='serial')
     return True
 
-# TODO: extract just the salt update part from `remaster`
-@requires_aws_stack
-def update_salt(stackname):
-    current_master_stack = core.find_master_for_stack(stackname)
-    return remaster(stackname, current_master_stack)
-
-# TODO: extract just the salt update part from `remaster`
-def update_salt_master(region=None):
-    "update the version of Salt installed on the master-server."
-    region = region or utils.find_region()
-    current_master_stackname = core.find_master(region)
-    return remaster(current_master_stackname, current_master_stackname)
-
-@requires_aws_stack
-def remaster_all(new_master_stackname):
+def remaster_all(*pname_list):
+    # same as the old master
```

Comment:
there should only be one master-server instance at a time
multiple masters is bad news. assumptions break and it gets complicated.